runtime/lua/hl_yank.lua

```diff
@@ -0,0 +1,29 @@
+--- Highlight the yanked region
+--
+--- use from init.vim via
+---   au TextYankPost * lua require'hl_yank'()
+--- customize highlight group and timeout via
+---   au TextYankPost * lua require'hl_yank'("IncSearch", 500)
+-- @param higroup highlight group for yanked region
+-- @param timeout time in ms before highlight is cleared
+-- @param event event structure
+return function(higroup, timeout, event)
+    local event = event or vim.v.event
+    if event.operator ~= 'y' or event.regtype == '' then return end
+    local higroup = higroup or "IncSearch"
+    local timeout = timeout or 500
+
+    local bufnr = vim.api.nvim_get_current_buf()
+    local yank_ns = vim.api.nvim_create_namespace('')
```

Comment:
Don't remember how this happened, but this ns got moved back into the function. This should get moved out of the function and created only once.

Reply:
No, this is on purpose. This creates a new namespace per yank, which in the end was a better UX than reusing a single namespace (when repeatedly yanking).

Reply:
nvm then :)